objectstorage/objectstorage.go

```diff
@@ -0,0 +1,123 @@
+package objectstorage
```

Comment:
I think that this package should have some errors declared as public to be managed by other package like the `api`. For example those used if the file does not exists, the file type is not supported or the objectID cannot be calculated.